src/components/checkbox/GroupableCheckboxReducersUtils.ts

```diff
@@ -0,0 +1,68 @@
+import * as _ from 'underscore';
+import {IReduxAction} from '../../utils/ReduxUtils';
+import {ICheckboxState} from './CheckboxReducers';
+import {IGroupableCheckboxActionPayload} from './GroupableCheckboxActions';
+import {groupableCheckboxInitialState, IGroupableCheckboxesState} from './GroupableCheckboxReducers';
+
+export const addParentCheckbox = (state: IGroupableCheckboxesState,
+                                  action: IReduxAction<IGroupableCheckboxActionPayload>): IGroupableCheckboxesState => {
+  const newCheckbox = {
+    id: action.payload.id,
+    checked: !!action.payload.checked,
+  };
+
+  if (state === groupableCheckboxInitialState) {
+    return {
+      ...state,
+      parentId: action.payload.id,
+      parent: newCheckbox,
+    };
+  }
+
+  state.parent = newCheckbox;
+  return state;
+};
+
+export const addChildCheckbox = (state: IGroupableCheckboxesState,
+                                 action: IReduxAction<IGroupableCheckboxActionPayload>): IGroupableCheckboxesState => {
+  const newChecboxes: ICheckboxState[] = state.checkboxes.concat({
+    id: action.payload.id,
+    checked: !!action.payload.checked,
+  });
+
+  const total: number =  state.total + 1;
+  const nbChecked: number = state.nbChecked + (action.payload.checked ? 1 : 0);
+
+  if (state === groupableCheckboxInitialState) {
+    return {
+      ...state,
+      parentId: action.payload.parentId,
+      total,
+      nbChecked,
+      checkboxes: newChecboxes,
+    };
+  }
+
+  return _.extend(state, {
+    checkboxes: newChecboxes,
+    total,
+    nbChecked,
+  });
+};
+
+export const removeParentCheckbox = (state: IGroupableCheckboxesState[],
+                                     action: IReduxAction<IGroupableCheckboxActionPayload>): IGroupableCheckboxesState[] => {
+  return _.reject(state, (checkboxState: IGroupableCheckboxesState) => {
+    return action.payload.id === checkboxState.parentId;
+  });
```

Comment:
You can omit the return statement and curly braces `return _.reject(state, (checkboxState: IGroupableCheckboxesState) => action.payload.id === checkboxState.parentId);`